dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fieldsmapper/FieldsParamMapper.java

```diff
@@ -31,27 +31,26 @@
 import java.util.List;
 import java.util.Map;
 
+import org.hisp.dhis.dxf2.events.AbstractParams;
```

Comment:
I really have some concerns on adding dependencies from `dxf2` here.
I really think we should have a deeper look at the first open point before we go through with this.
I would really like that when we finally decide to remove the old tracker we can just remove one package (obviously this is not gonna happen), or anyway avoid the need to go through the code and remove small bits here and there.

So the question here is really if it make sense to have all common parts in the `dhis-api` module or if it even make sense to duplicate some code, leaving the old one as it is and move the new one closer to the web

Reply:
Please also know that we are hoping to remove the `dxf2` module in the future. When the old tracker importer is gone, we want this module to only contain metadata importer code.

Reply:
@teleivo @enricocolasante @stian-sandvold I see there is some confusion here. The `Params` classes were already in the dxf2 package so we are not adding dependencies here. The `Params` are used by the export endpoints using services all the way through the `dxf2` package. So I am not sure what is the best solution but duplicating the code or moving to the web part what does it mean? These are classes used in the services so in my opinion if we just move this to the `dhis-api` then later on when we get rid of the old tracker, we can move this stuff into the tracker module. If we duplicate code now we just have two things to maintain

Reply:
Hey @lucaCambi77 ,

in my case, I am referring to the files under the dxf2 package. I would rather we don't change those, and rather create duplicate versions of them, to use in the new tracker module. That way we have less dependencies in dxf2 when we do remove the old tracker code from there.

As it stands, any work you do in the dxf2 package for master, will be removed in 2.41 most likely, if it relates to import at least, and probably to export as well, since the old endpoints for export will no longer be needed with the old importer gone.

Reply:
The new tracker export is totally dependent on the dxf2 package see [here](https://github.com/dhis2/dhis2-core/blob/eb48b16bc219132cd877edec3428d402f78bdc59/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/TrackerEnrollmentsExportController.java#L44) for enrollments, here for [tracked entities](https://github.com/dhis2/dhis2-core/blob/6c3a535f28562b12bc7803ce8c102a9cbd86f982/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/TrackerTrackedEntitiesExportController.java#L54) . So i think first we should  think about how to deal with this dependency

Reply:
So what do we want to do with this PR? Shall we close it?